app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+            if(strlen($uniqueMainKeyword) > 0) {
+                // Insert new main keyword
+                $mainKeyword = new Keywords();
+                $mainKeyword->keyword = $uniqueMainKeyword;
+                $mainKeyword->type = 1;
+                $mainKeyword->save();
+            }
+        }
+
+        $postMetaMkw = PostMeta::getPostMeta($post->id, isset($mainKeyword->id) ? $mainKeyword->id : 0);
+        if(empty($postMetaMkw) && isset($mainKeyword->id)) {
+            // Insert new post-keyword relationship if not exist
+            $postMetaMkw = new PostMeta();
+            $postMetaMkw->post_id = $post->id;
+            $postMetaMkw->keyword_id = $mainKeyword->id;
+            $postMetaMkw->save();
+        }
+
+        // Sub keywords
+        $subs = explode(',', $request->get('subkws'));
+        foreach ($subs as $item) {
+            $trimmedItem = trim($item);
+            if(strlen($trimmedItem) == 0) {
+                continue;
+            }
+
+            $subKeyword = Keywords::getKeyword($trimmedItem, 2);
+
+            if(empty($subKeyword)) {
+                // Insert new keyword
+                $subKeyword = new Keywords();
+                $subKeyword->keyword = $trimmedItem;
+                $subKeyword->type = 2;
+                $subKeyword->save();
+
+                // Save post-keyword relationship
+            }
+
+            $postMetaSkw = PostMeta::getPostMeta($post->id, $subKeyword->id);
+            if(empty($postMetaSkw)) {
+                // Insert new post-keyword relationship if not exist
+                $postMetaSkw = new PostMeta();
+                $postMetaSkw->post_id = $post->id;
+                $postMetaSkw->keyword_id = $subKeyword->id;
+                $postMetaSkw->save();
+            }
+
+
+        }
+
+        if(strpos($request->get('submit_type'),'公開') !== false || strpos($request->get('submit_type'),'下書き') !== false){
+            return redirect()->to(action('Webapp\\PostController@getEdit', $post->id))->send();
+        }else if(strpos($request->get('submit_type'),'プレビュー') !== false){
+            return redirect()->to(action('Webapp\\PostController@getPreviewIframe', $post->id))->send();
+        }
+
+        return redirect()->to(action('Webapp\\PostController@getEdit', $post->id))->send();
+    }
+
+    public function postCreateAjax(Request $request) {
+        $post = new Post();
+
+        if(($postId = $request->get('post_id', 0)) != 0) {
+            $post = (!empty($postTmp = Post::find($postId)) && $postTmp->user_id == $this->currentUser->id) ? $postTmp : $post;
+        }
+
+        $postTitle = $request->get('post_title');
+        $postContent = $request->get('post_content');
+        $postChecksum = $request->get('checksum', 0);
+        $postDescription = trim($request->get('post_description'));
+
+        if($post->isLock($this->currentSession)) {
+            $this->activityLog->setInfo('Locked');
+            return ['error' => [
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "error" 15 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)